freqtrade/freqai/freqai_interface.py

```diff
@@ -363,10 +364,10 @@ def start_live(
         # load the model and associated data into the data kitchen
         self.model = self.dd.load_data(metadata["pair"], dk)
 
-        with self.analysis_lock:
-            dataframe = self.dk.use_strategy_to_populate_indicators(
-                strategy, prediction_dataframe=dataframe, pair=metadata["pair"]
-            )
+        dataframe = dk.use_strategy_to_populate_indicators(
+            strategy, prediction_dataframe=dataframe, pair=metadata["pair"],
+            do_corr_pairs=self.get_corr_dataframes
+        )
```

Comment:
we no longer use  `analysis_lock` here ... does that mean we can/could remove that lock completely? (in my understanding, the problem we tried to solve here was fixed in ta-lib)

or is it still necessary in some other place? (it's used once more in `extract_data_and_train_model()`).

Reply:
Sure, we can add the analysis lock removal to this PR.